Sources/Sentry/SentryTracer.m

```diff
@@ -408,19 +413,19 @@ - (void)canBeFinished
     if (self.rootSpan.isFinished)
         return;
 
-    BOOL hasChildrenToWaitFor = [self hasChildrenToWaitFor];
-    if (self.isWaitingForChildren == NO && !hasChildrenToWaitFor && [self hasIdleTimeout]) {
+    BOOL hasChildSpansToWaitFor = [self hasUnfinishedChildSpansAndNeedToWaitUntilTheyAreFinished];
+    if (self.wasFinishCalled == NO && !hasChildSpansToWaitFor && [self hasIdleTimeout]) {
         [self dispatchIdleTimeout];
         return;
     }
 
-    if (!self.isWaitingForChildren || hasChildrenToWaitFor)
+    if (!self.wasFinishCalled || hasChildSpansToWaitFor)
```

Comment:
This is becoming more clear now with `isWaitingForChildren`->`wasFinishCalled` 👍🏻